src/state.ts

```diff
@@ -0,0 +1,24 @@
+import {Event} from './event';
+
+export class State {
+	public readonly events;
+
+	constructor() {
+		this.events = {};
```

Comment:
if events is the class events this line should invoke the constructor as in new Events(…) rather than assigning to an object